File: Doubly_linked_list/DoublyLinkedListCreateNNodes.java
// File Name: DoublyLinkedListCreateNNodes.java
// Program to create a doubly linked list with n nodes

import java.util.Scanner;

class Node {
    int data;
    Node prev;
    Node next;

    Node(int data) {
        this.data = data;
        this.prev = null;
        this.next = null;
    }
}

public class DoublyLinkedListCreateNNodes {
    static Node head = null;

    // Function to create doubly linked list with n nodes
    static void createList(int n) {
        Scanner sc = new Scanner(System.in);
        Node temp = null;

        for (int i = 1; i <= n; i++) {
            System.out.print("Enter data for node " + i + ": ");
            int value = sc.nextInt();
            Node newNode = new Node(value);

            if (head == null) {
                head = newNode;
                temp = head;
            } else {
                temp.next = newNode;
                newNode.prev = temp;
                temp = newNode;
            }
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter number of nodes: ");
        int n = sc.nextInt();

        createList(n);
    }
}

File: Doubly_linked_list/doubly_linked_list_create_n_nodes.c
// File Name: doubly_linked_list_create_n_nodes.c
// Program to create a doubly linked list with n nodes

#include <stdio.h>
#include <stdlib.h>

// Definition of node
struct node {
    int data;
    struct node *prev;
    struct node *next;
};

struct node *head = NULL;

// Function to create doubly linked list with n nodes
void create_list(int n) {
    struct node *newnode, *temp;
    int i, value;

    for (i = 1; i <= n; i++) {
        printf("Enter data for node %d: ", i);
        scanf("%d", &value);

        newnode = (struct node *)malloc(sizeof(struct node));
        newnode->data = value;
        newnode->prev = NULL;
        newnode->next = NULL;

        if (head == NULL) {
            head = newnode;
            temp = head;
        } else {
            temp->next = newnode;
            newnode->prev = temp;
            temp = newnode;
        }
    }
}

int main() {
    int n;
    printf("Enter number of nodes: ");
    scanf("%d", &n);

    create_list(n);
    return 0;
}
